src/InputBar.js

```diff
@@ -0,0 +1,29 @@
+import { useState } from "react";
+
+export default function InputBar({ onNameChange }) {
+  const [name, setName] = useState([""]);
+
+  function handleOnChange(e) {
+    setName(e.target.value);
+  }
+  function handleSubmit(e) {
+    e.preventDefault();
+    onNameChange(name);
+    e.target.reset();
+  }
+  return (
+    <form onSubmit={handleSubmit}>
+      <input
+        className="app__input"
+        text="text"
+        name="name"
+        placeholder="Type name..."
+        onChange={handleOnChange}
+        value={name}
+      />
+      <p>
+        <button>Submit</button>
+      </p>
+    </form>
+  );
+}
```

Comment:
Amazing job, you made the component, you passed a callback prop `onNameChange`. Really good.

The only thing I could think about change is the name, maybe not `InputBar` but `InputForm` or something like this. 

But I see good stuff in here!